lib/matplotlib/backend_bases.py

```diff
@@ -508,30 +508,31 @@ def get_image_magnification(self):
         """
         return 1.0
 
-    def draw_image(self, gc, x, y, im, trans=None):
+    def draw_image(self, gc, x, y, im, transform=None):
         """
-        Draw the image instance into the current axes;
+        Draw an RGBA image.
 
         *gc*
-            a GraphicsContext containing clipping information
+            a :class:`GraphicsContextBase` instance with clipping information.
 
         *x*
-            is the distance in pixels from the left hand side of the canvas.
+            the distance in pixels from the left hand side of the canvas.
 
         *y*
-            the distance from the origin.  That is, if origin is
-            upper, y is the distance from top.  If origin is lower, y
-            is the distance from bottom
+            the distance in pixels from the bottom side of the canvas.
```

Comment:
Explanation: `draw_image()` does not know anything about the `origin` property of the `AxesImage` or `FigureImage` instance being drawn, because it doesn't get a reference to the image instance, just the image data. Hence, the meaning of `y` is independent of `origin`.